extensions/dialog/common/client.mjs

```diff
@@ -95,8 +96,8 @@ class DialogModal extends Modal {
       return btnList;
     });
 
-    if (result === false) return {cancelled: true};
-    return {token: result.dataset.token};
+    if (result) return result.dataset.token;
```

Comment:
The primary reason for returning an object is so that we can differentiate between various reasons why the no choice was picked. One possible reason is because another Modal is already open, or the client disconnected, or if some error is encountered during RPC (i.e. timeout).

Currently the s2c rpc system returns an object with `error` key when something happens. A better way might be to have a system that passes exception across the RPC, but that'll take time to implement.

I'm OK with the current code change if you're sure there'll be no need to distinguish between various failure types.

Reply:
Well, I'm not sure that there will be no need to distinguish between various failure types.
I will revert this modification.